packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -73,14 +77,16 @@ export default createRule<Options, MessageId>({
     messages: {
       voidReturnArgument:
         'Promise returned in function argument where a void return was expected.',
-      voidReturnVariable:
-        'Promise-returning function provided to variable where a void return was expected.',
+      voidReturnAttribute:
+        'Promise-returning function provided to attribute where a void return was expected.',
       voidReturnProperty:
         'Promise-returning function provided to property where a void return was expected.',
       voidReturnReturnValue:
         'Promise-returning function provided to return value where a void return was expected.',
-      voidReturnAttribute:
-        'Promise-returning function provided to attribute where a void return was expected.',
+      voidReturnSubtype:
+        "Promise-returning method provided where a void return was expected by heritage type '{{ heritageTypeName }}'.",
+      voidReturnVariable:
+        'Promise-returning function provided to variable where a void return was expected.',
```

Comment:
Other enumerations over the `checksVoidReturn` options were sorted alphabetically, so just went ahead and did a swap here (attribute <=> variable) to maintain that consistent order